jfrog-cli/artifactory/commands/curl/curl.go

```diff
@@ -0,0 +1,224 @@
+
+	return
+}
+
+// Get --server-id flag value from the command, and remove it.
+func (curlCmd *CurlCommand) getAndRemoveServerIdFromCommand() (string, error) {
+	// Get server id.
+	serverIdFlagIndex, serverIdValueIndex, serverIdValue, err := curlCmd.findFlag("--server-id")
+	if err != nil {
+		return "", err
+	}
+
+	// Remove --server-id from Command if required.
+	if serverIdFlagIndex != -1 {
+		curlCmd.Arguments = append(curlCmd.Arguments[:serverIdFlagIndex], curlCmd.Arguments[serverIdValueIndex+1:]...)
+	}
+
+	return serverIdValue, nil
+}
+
+// Find an argument in Command.
+// A command flag is prefixed by '-' or '--'.
+// Use this method ONLY after removing all JFrog-CLI flags, i.e flags in the form: '--my-flag=value' are not allowed.
+// An argument is any provided candidate which is not a flag or a flag value.
+func (curlCmd *CurlCommand) findNextArg() (int, string) {
+	skipThisArg := false
+	for index, arg := range curlCmd.Arguments {
+		// Check if shouldn't check current arg.
+		if skipThisArg {
+			skipThisArg = false
+			continue
+		}
+
+		// If starts with '--', meaning a flag which its value is at next slot.
+		if strings.HasPrefix(arg, "--") {
+			skipThisArg = true
+			continue
+		}
+
+		// Check if '-'.
+		if strings.HasPrefix(arg, "-") {
+			if len(arg) > 2 {
+				// Meaning that this flag also contains its value.
+				continue
+			}
+			// If reached here, means that the flag value is at the next arg.
+			skipThisArg = true
+			continue
+		}
+
+		// Found an argument
+		return index, arg
+	}
+
+	// If reached here, didn't find an argument.
+	return -1, ""
+}
+
+// Find value of required flag in Command.
+// If flag does not exist, returned indexes are -1 and error is nil.
+// Return values:
+// err - error if flag exists but failed to extract its value.
+// flagIndex - index of flagName in Command.
+// flagValueIndex - index in Command in which the value of the flag exists.
+// flagValue - value of flagName.
+func (curlCmd *CurlCommand) findFlag(flagName string) (flagIndex, flagValueIndex int, flagValue string, err error) {
+	flagIndex = -1
+	flagValueIndex = -1
+	for index, arg := range curlCmd.Arguments {
+		// Check current argument.
+		if !strings.HasPrefix(arg, flagName) {
+			continue
+		}
+
+		// Get flag value.
+		flagValue, flagValueIndex, err = curlCmd.getFlagValueAndValueIndex(flagName, index)
+		if err != nil {
+			return
+		}
+
+		// If was not the correct flag, continue looking.
+		if flagValueIndex == -1 {
+			continue
+		}
+
+		// Return value.
+		flagIndex = index
+		return
+	}
+
+	// Flag not found.
+	return
+}
+
+// Get the provided flag's value, and the index of the value.
+// Value-index can either be same as flag's index, or the next one.
+// Return error if flag is found, but couldn't extract value.
+// If the provided index doesn't contain the searched flag, return flagIndex = -1.
+func (curlCmd *CurlCommand) getFlagValueAndValueIndex(flagName string, flagIndex int) (string, int, error) {
```

Comment:
I think this can be simplified that you will not need to pass the flagIndex and the logic will be in findFlag. This can reduce the amount of returned objects.

Reply:
Added returned values names.

I don't think that not passing the flagIndex simplifies things in any way.
As the flag itself is at some index, its value may be located either in the same index or the next. getFlagValueAndValueIndex finds the value, depending on the information it gets of the flag index.